metricbeat/module/kubernetes/state_container/state_container.go

```diff
@@ -59,14 +59,15 @@ var (
 				},
 			)),
 
-			"kube_pod_container_status_ready":                  p.BooleanMetric("status.ready"),
-			"kube_pod_container_status_restarts_total":         p.Metric("status.restarts"),
-			"kube_pod_container_status_running":                p.KeywordMetric("status.phase", "running"),
-			"kube_pod_container_status_terminated":             p.KeywordMetric("status.phase", "terminated"),
-			"kube_pod_container_status_waiting":                p.KeywordMetric("status.phase", "waiting"),
-			"kube_pod_container_status_terminated_reason":      p.LabelMetric("status.reason", "reason"),
-			"kube_pod_container_status_waiting_reason":         p.LabelMetric("status.reason", "reason"),
-			"kube_pod_container_status_last_terminated_reason": p.LabelMetric("status.last_terminated_reason", "reason"),
+			"kube_pod_container_status_ready":                     p.BooleanMetric("status.ready"),
+			"kube_pod_container_status_restarts_total":            p.Metric("status.restarts"),
+			"kube_pod_container_status_running":                   p.KeywordMetric("status.phase", "running"),
+			"kube_pod_container_status_terminated":                p.KeywordMetric("status.phase", "terminated"),
+			"kube_pod_container_status_waiting":                   p.KeywordMetric("status.phase", "waiting"),
+			"kube_pod_container_status_terminated_reason":         p.LabelMetric("status.reason", "reason"),
+			"kube_pod_container_status_waiting_reason":            p.LabelMetric("status.reason", "reason"),
+			"kube_pod_container_status_last_terminated_reason":    p.LabelMetric("status.last_terminated_reason", "reason"),
+			"kube_pod_container_status_last_terminated_timestamp": p.Metric("status.last_terminated_reason.sec"),
```

Comment:
Here you map `kube_pod_container_status_last_terminated_timestamp`  ksm metric to elasticsearch field `kubernetes.container.status.last_terminated_reason.sec`.  But then , the field you declare in the fields.yaml is `last_terminated_reason_timestamp`. Am I missing something?

Reply:
my mistake, reverted this - https://github.com/elastic/beats/pull/39200/commits/180afdf5df6fe200849b8575a3ec9ca5af541a91

It is a leftover, I was considering other naming, not really sure regarding the name `last_terminated_reason_timestamp`. For example in state_cronjob - https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_cronjob/state_cronjob.go#L33-L34 is used `*.sec` for time metrics, but then I figured out that there is not really a pattern for time metrics, for example:
[job](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_job/state_job.go#L39-L40) - `time.*`  (type of the field: data)
[statefulset](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_statefulset/state_statefulset.go#L30) - no `time` or `sec` in name  (type: long)
[storageclass](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_storageclass/state_storageclass.go#L37) - no `time` or `sec` in name (type: date)
[namespace](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_namespace/state_namespace.go#L30) - `created.sec`, type: double

do you have any thoughts on naming?

Reply:
done in https://github.com/elastic/beats/pull/39200/commits/90d82b61952bd251c1436d0d362b8e08cac62697